backend/src/apis/address_ownership.rs

```diff
@@ -1,29 +1,23 @@
-use crate::contracts::{
-    generated::summa_contract::AddressOwnershipProof,
-    signer::{AddressInput, SummaSigner},
-};
+use crate::contracts::{generated::summa_contract::AddressOwnershipProof, signer::SummaSigner};
 use std::{error::Error, result::Result};
 
 use super::csv_parser::parse_signature_csv;
 
-pub struct AddressOwnership {
+pub struct AddressOwnership<'a> {
     address_ownership_proofs: Vec<AddressOwnershipProof>,
-    signer: SummaSigner,
+    signer: &'a SummaSigner,
 }
```

Comment:
I understand (and appreciate) your choice of using a lifetime annotation for the `SummaSigner` reference. This is the only place in the code base we use that. Do you think there are other places in which this might be needed for managing references?

Reply:
The lifetime annotation is necessary because the `signer` instance is shared between both `Round` and `AddressOwnership`. I traced the flow of data like this: `Round` → `Snapshot` → `MerkleSumTree` → `Node`, `Entry`. There doesn't seem to be any need for data sharing along that chain.

The only improvement I identified is that there's no need to clone `merkle_sum_tree` when initializing circuits.
For instance, instead of 
`let mut circuit = MstInclusionCircuit::<LEVELS, N_ASSETS, N_BYTES>::init(merkle_sum_tree.clone(), 0);`, 
it could be 
`let mut circuit = MstInclusionCircuit::<LEVELS, N_ASSETS, N_BYTES>::init(&merkle_sum_tree, 0);`.

Reply:
Good point! Gonna create an issue for that

Reply:
> Good point! Gonna create an issue for that

I got fixed one, let me assign to me :)